subsys/bluetooth/audio/has_client.c

```diff
@@ -82,11 +88,179 @@ static enum bt_has_capabilities get_capabilities(const struct has_inst *inst)
 	return caps;
 }
 
+static void handle_read_preset_rsp(struct has_inst *inst, struct net_buf_simple *buf)
+{
+	const struct bt_has_cp_read_preset_rsp *pdu;
+	struct bt_has_preset_record record;
+	char name[BT_HAS_PRESET_NAME_MAX + 1]; /* + 1 byte for null-terminator */
+	size_t name_len;
+
+	BT_DBG("conn %p buf %p", (void *)inst->conn, buf);
+
+	if (buf->len < sizeof(*pdu)) {
+		BT_ERR("malformed PDU");
+		return;
+	}
+
+	pdu = net_buf_simple_pull_mem(buf, sizeof(*pdu));
+
+	if (pdu->is_last > BT_HAS_IS_LAST) {
+		BT_WARN("unexpected is_last value 0x%02x", pdu->is_last);
+	}
+
+	record.index = pdu->index;
+	record.properties = pdu->properties;
+	record.name = name;
+
+	name_len = buf->len + 1; /* + 1 byte for NULL terminator */
+	if (name_len > ARRAY_SIZE(name)) {
+		BT_WARN("name is too long (%zu > %u)", buf->len, BT_HAS_PRESET_NAME_MAX);
+
+		name_len = ARRAY_SIZE(name);
+	}
+
+	utf8_lcpy(name, pdu->name, name_len);
+
+	client_cb->preset_read_rsp(&inst->has, 0, &record, !!pdu->is_last);
+}
+
+static void handle_generic_update(struct has_inst *inst, struct net_buf_simple *buf, bool is_last)
+{
+	const struct bt_has_cp_generic_update *pdu;
+	struct bt_has_preset_record record;
+	char name[BT_HAS_PRESET_NAME_MAX + 1]; /* + 1 byte for null-terminator */
+	size_t name_len;
+
+	if (buf->len < sizeof(*pdu)) {
+		BT_ERR("malformed PDU");
+		return;
+	}
+
+	pdu = net_buf_simple_pull_mem(buf, sizeof(*pdu));
+
+	record.index = pdu->index;
+	record.properties = pdu->properties;
+	record.name = name;
+
+	name_len = buf->len + 1; /* + 1 byte for NULL terminator */
+	if (name_len > ARRAY_SIZE(name)) {
+		BT_WARN("name is too long (%zu > %u)", buf->len, BT_HAS_PRESET_NAME_MAX);
+
+		name_len = ARRAY_SIZE(name);
+	}
+
+	utf8_lcpy(name, pdu->name, name_len);
+
+	client_cb->preset_update(&inst->has, pdu->prev_index, &record, is_last);
+}
+
+static void handle_preset_deleted(struct has_inst *inst, struct net_buf_simple *buf, bool is_last)
+{
+	if (buf->len < sizeof(uint8_t)) {
+		BT_ERR("malformed PDU");
+		return;
+	}
+
+	client_cb->preset_deleted(&inst->has, net_buf_simple_pull_u8(buf), is_last);
```

Comment:
Suggestion for improved readability: 

```suggestion
	uint8_t index;
	
	if (buf->len < sizeof(index {
		BT_ERR("malformed PDU");
		return;
	}
	
	index = net_buf_simple_pull_u8(buf);

	client_cb->preset_deleted(&inst->has, index, is_last);
```

(applies multiple places).